File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/location/PulsingLocationCircleAnimator.java
package org.maplibre.android.location;

/**
 * Manages the logic of the interpolated animation which is applied to the LocationComponent's pulsing circle
 */
public class PulsingLocationCircleAnimator extends MapLibreFloatAnimator {

  /**
   *
   * @param updateListener  the {@link AnimationsValueChangeListener} associated with this animator.
   * @param maxAnimationFps the maximum frames per second that the animator should use. Default
   *                        is the {@link LocationAnimatorCoordinator#maxAnimationFps} variable.
   */
  public PulsingLocationCircleAnimator(AnimationsValueChangeListener updateListener,
                                       int maxAnimationFps,
                                       float circleMaxRadius) {
    super(new Float[]{0f, circleMaxRadius}, updateListener, maxAnimationFps);
  }
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/location/PulsingLocationCircleAnimator.kt
package org.maplibre.android.location

import org.maplibre.android.location.MapLibreAnimator.AnimationsValueChangeListener

/**
 * Manages the logic of the interpolated animation which is applied to the LocationComponent's pulsing circle
 *
 * @param updateListener  the [AnimationsValueChangeListener] associated with this animator.
 * @param maxAnimationFps the maximum frames per second that the animator should use. Default
 *                        is the [LocationAnimatorCoordinator.maxAnimationFps] variable.
 */
internal class PulsingLocationCircleAnimator(
    updateListener: AnimationsValueChangeListener<Float>,
    maxAnimationFps: Int,
    circleMaxRadius: Float,
) : MapLibreFloatAnimator(arrayOf(0f, circleMaxRadius), updateListener, maxAnimationFps)
